src/back/space/src/api/invitation.cpp

```diff
@@ -0,0 +1,298 @@
+#include "../service/service.hpp"
+#include "../../../shared/headers.hpp"
+#include "../../../shared/errors.hpp"
+#include "../../../shared/paging.hpp"
+#include "../../../shared/paging.сpp"
+#include "../model/role.hpp"
+#include "../model/invitation_serialize.hpp"
+
+namespace svetit::space::handlers {
+
+Invitation::Invitation(
+	const components::ComponentConfig& conf,
+	const components::ComponentContext& ctx)
+	: server::handlers::HttpHandlerJsonBase{conf, ctx}
+	, _s{ctx.FindComponent<Service>()}
+{}
+
+formats::json::Value Invitation::GetList(
+	const server::http::HttpRequest& req,
+	const formats::json::Value& body) const
+{
+	formats::json::ValueBuilder res;
+
+	try {
+		auto paging = parsePaging(req);
+		res["list"] = _s.GetInvitationList(paging.start, paging.limit);
+		res["total"] = _s.GetInvitationsCount();
+	}
+	catch(const errors::BadRequest& e) {
+		res["err"] = e.what();
+		req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+		return res.ExtractValue();
+	}
+	catch(const std::exception& e) {
+		LOG_WARNING() << "Fail to get invitations list: " << e.what();
+		req.SetResponseStatus(server::http::HttpStatus::kInternalServerError);
+		res["err"] = "Fail to get invitations list";
+	}
+
+	return res.ExtractValue();
+}
+
+formats::json::Value Invitation::Post(
+	const server::http::HttpRequest& req,
+	const formats::json::Value& body) const
+{
+	formats::json::ValueBuilder res;
+
+	const auto& creatorId = req.GetHeader(headers::kUserId);
+	if (creatorId.empty()) {
+		res["err"] = "Access denied";
+		req.SetResponseStatus(server::http::HttpStatus::kUnauthorized);
+		return res.ExtractValue();
+	}
+
+	const auto link = req.GetArg("link");
+	bool linkMode = false;
+	model::SpaceInvitation invitation;
+
+	if (req.HasArg("link")) {
+		if (!link.empty()) {
+			linkMode = true;
+		} else {
+			req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+			res["err"] = "Link must not be empty";
+			return res.ExtractValue();
+		}
+	} else {
+		invitation = body.As<model::SpaceInvitation>();
+
+		if (invitation.spaceId.is_nil() || invitation.userId.empty()) {
+			req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+			res["err"] = "Params must be set";
+			return res.ExtractValue();
+		}
+
+		if (!_s.ValidateRole(invitation.role)) {
+			req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+			res["err"] = "Wrong role";
+			return res.ExtractValue();
+		}
+	}
+
+	try {
+		if (linkMode) {
+			if (_s.InviteByLink(creatorId, link)) {
+				req.SetResponseStatus(server::http::HttpStatus::kCreated);
+			} else {
+				req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+				res["err"] = "Wrong link";
+			}
+		} else {
+			if (_s.Invite(creatorId, invitation.spaceId, invitation.userId, invitation.role)) {
+				req.SetResponseStatus(server::http::HttpStatus::kCreated);
+			} else {
+				req.SetResponseStatus(server::http::HttpStatus::kBadRequest);
+				res["err"] = "Can't create invite";
+			}
+		}
```

Comment:
Попробуй пожалуйста в других местах в подобном стиле сделать: 👇

```suggestion
	try {
		if (req.HasArg("link")) {
			const auto link = req.GetArg("link");
			if (link.empty())
				throw errors::BadRequest("empty linkId");
			if (!_s.InviteByLink(creatorId, link))
				throw std::runtime_error("failed invite by link");
			req.SetResponseStatus(server::http::HttpStatus::kCreated);
			return res.ExtractValue();
		}
		
		auto invitation = body.As<model::SpaceInvitation>();

		if (invitation.spaceId.is_nil() || invitation.userId.empty())
			throw errors::BadRequest("Params must be set");

		if (!_s.ValidateRole(invitation.role))
			throw errors::BadRequest("Wrong role");

		if (!_s.Invite(creatorId, invitation.spaceId, invitation.userId, invitation.role))
			throw std::runtime_error("failed invite");

		req.SetResponseStatus(server::http::HttpStatus::kCreated);
```

Reply:
Готово (переписал так во всех обработчиках).